db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2237,9 +2229,18 @@ Status DBImpl::AtomicFlushMemTables(
     return Status::TryAgain(oss.str());
   }
   Status s;
+  autovector<ColumnFamilyData*> candidate_cfds;
+  if (provided_candidate_cfds.empty()) {
+    for (ColumnFamilyData* cfd : *versions_->GetColumnFamilySet()) {
+      candidate_cfds.push_back(cfd);
+    }
```

Comment:
I think we're supposed to hold DB mutex for iterating the column families.

Reply:
Probably need to reference the CFDs too although that was missing from the existing code. This looks like a proper usage to me - https://github.com/facebook/rocksdb/blob/d053926fa2830598a5f6d8c1a7b84ee7852ef9af/db/db_impl/db_impl.cc#L5671-L5679

Reply:
Thanks! Second time I made this mistake - https://github.com/facebook/rocksdb/pull/11041#discussion_r1052738511 I really need to pause before using anything in versionset/version/vstorage to see if a lock is needed!!!!

Reply:
Fixed this comment - also took care of the unref as well as a similar case in SelectColumnFamiliesForAtomicFlush